### PR TITLE
WIP: Use binary crate

### PR DESCRIPTION
**Pull request by kenz-gelsoft**
- `cargo build --features use_binary` builds with prebiult repo's wx binary if specified for that target.
- ~~doesn't link currently~~ Now it builds and run locally with help from @ancwrd1.
- Adding GitHub Actions for these configuration will be another Issue and PR,

### CODE REVIEW

**Comment by kenz-gelsoft** [comment]
@ancwrd1 I think with this PR, you can use your vendored repo with `--features use_binary` flag.

And also it is able to use another binary crate by [overriding concrete crates in consumer project's cargo configuration](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html) like:
```toml
[dependencies]
wx-x86_64-pc-windows-gnu = { git = "another/git/repo" }
```

This will support your suggested use case in #72? Any other feedbacks?

Thanks in advance.

File: wx-base/Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,18 @@ edition = "2021"
...
+
+[target.'cfg(target_os = "macos")'.dependencies]
+wx-universal-apple-darwin = { git = "https://github.com/ancwrd1/wx-universal-apple-darwin" }
+
+[target.x86_64-pc-windows-msvc.dependencies]
+wx-x86_64-pc-windows-msvc = { git = "https://github.com/kenz-gelsoft/wx-x86_64-pc-windows-msvc" }
+
+[target.x86_64-pc-windows-gnu.dependencies]
+wx-x86_64-pc-windows-gnu = { git = "https://github.com/ancwrd1/wx-x86_64-pc-windows-gnu" }
+
```
[the remark this answers is not in the document]
Comment (kenz-gelsoft) [reply wx-base/Cargo.toml:24]: done

File: wx-core/Cargo.toml [hunk 2]
```diff
@@ -5,9 +5,21 @@ edition = "2021"
...
 wx-base = { path = "../wx-base" }
 
+[target.'cfg(target_os = "macos")'.dependencies]
+wx-universal-apple-darwin = { git = "https://github.com/ancwrd1/wx-universal-apple-darwin" }
+
+[target.x86_64-pc-windows-msvc.dependencies]
+wx-x86_64-pc-windows-msvc = { git = "https://github.com/kenz-gelsoft/wx-x86_64-pc-windows-msvc" }
+
+[target.x86_64-pc-windows-gnu.dependencies]
+wx-x86_64-pc-windows-gnu = { git = "https://github.com/ancwrd1/wx-x86_64-pc-windows-gnu" }
```
[the remark this answers is not in the document]
Comment (kenz-gelsoft) [reply wx-core/Cargo.toml]: done

**Comment by ancwrd1** [comment]
Thanks for your work! I think the common feature name used by many popular crates is 'vendored' (or like 'vendored-wx', etc). For example it is used in native-tls, openssl, reqwest. But it's not too important.
I see that you have also added `-msvc` based binaries, that's nice! I will try it.

**Comment by kenz-gelsoft** [comment]
> I think the common feature name used by many popular crates is 'vendored' (or like 'vendored-wx', etc).

OK, to rename to that.

> I see that you have also added `-msvc` based binaries, that's nice! I will try it.

I'm afraid that it is not what you expected. The `-msvc` binary is wxWidgets official DLL distribution for MSVC which packaged in compatible way with your prebuilt crates. So, they are not static linked and requires you place DLLs in same directory of executables manually (at least, for now).

It's not hard to copy corresponding DLLs from `-msvc` to target build directory. I'll work on this in part of #34.

**Comment by ancwrd1** [comment]
> I'm afraid that it is not what you expected. The `-msvc` binary is wxWidgets official DLL distribution for MSVC which packaged in compatible way with your prebuilt crates. So, they are not static linked and requires you place DLLs in same directory of executables manually (at least, for now).

I've now added static manually built lean and mean MSVC repo as well:
https://github.com/ancwrd1/wx-x86_64-pc-windows-msvc.git

The following flags are needed for compilation: `/std:c++14`, `/EHsc`
`/D_CRT_SECURE_NO_WARNINGS` can be also specified to avoid warnings.

**Comment by kenz-gelsoft** [comment]
@ancwrd1 Thank you for providing static prebuilt crate.

I'll utilize that when I'll start writing unit tests for windows in future.

**Pull request merged**